### PR TITLE
feat: hand raise browser notification when away from call

### PR DESCRIPTION
**Pull request by eswarclynn**
### Details(context, link the issue, how was the bug fixed, what does the new feature do)

-
-

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx [hunk 1]
```diff
@@ -116,6 +118,7 @@ const PreviewJoin = ({
         savePreferenceAndJoin();
       } else {
         preview();
+        requestPermission();
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Preview/PreviewJoin.tsx:121]: should we do this after preview.then? when does the notification show now?

File: packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx [hunk 2]
```diff
@@ -38,8 +42,12 @@ export const HandRaisedNotifications = () => {
       const showCTA = peer?.roleName && (on_stage_exp?.off_stage_roles || [])?.includes(peer.roleName);
       ToastBatcher.showToast({ notification, type: showCTA ? 'RAISE_HAND_HLS' : 'RAISE_HAND' });
       console.debug('Metadata updated', notification.data);
+
+      showNotification(`${peer?.name} raised hand`, {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:46]: what if there are 10-20 notifications?

File: packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx [hunk 3]
```diff
@@ -38,8 +44,14 @@ export const HandRaisedNotifications = () => {
       const showCTA = peer?.roleName && (on_stage_exp?.off_stage_roles || [])?.includes(peer.roleName);
       ToastBatcher.showToast({ notification, type: showCTA ? 'RAISE_HAND_HLS' : 'RAISE_HAND' });
       console.debug('Metadata updated', notification.data);
+
+      if (amIScreenSharing) {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:48]: and document.hidden as well

File: packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx [hunk 4]
```diff
@@ -41,5 +50,30 @@ export const HandRaisedNotifications = () => {
...
+        return;
+      }
+
+      // Don't show toast message in case of local peer.
+      if (roomState !== HMSRoomState.Connected || notification.data.isLocal || !isSubscribing) {
+        return;
+      }
+
+      const hasPeerHandRaised = vanillaStore.getState(selectHasPeerHandRaised(notification.data.id));
+      const peer = vanillaStore.getState(selectPeerByID(notification.data.id));
```
[the remark this answers is not in the document]
Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:65]: name should be there no?

File: packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx [hunk 5]
```diff
@@ -41,5 +50,30 @@ export const HandRaisedNotifications = () => {
...
+      const handRaisedPeers = vanillaStore.getState(selectHandRaisedPeers);
+      if (amIScreenSharing && hasPeerHandRaised) {
+        const title = `${peer?.name} ${
+          handRaisedPeers.length > 1 ? `and ${handRaisedPeers.length - 1} others` : ''
+        } raised hand`;
+        console.log(title);
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:71]: can be removd

File: packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (eswarclynn) [reply packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:65]: Copied code from useEffect above, would data still be there if peer has left?

[the remark this answers is not in the document]
Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/components/Notifications/HandRaisedNotifications.tsx:65]: it should be there.

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**